tchannel-example/src/main/java/com/uber/tchannel/hyperbahn/HyperbahnExample.java

```diff
@@ -69,12 +69,15 @@ public static void main(String[] args) throws Exception {
         future.addCallback(new TFutureCallback<JsonResponse<AdvertiseResponse>>() {
             @Override
             public void onResponse(JsonResponse<AdvertiseResponse> response) {
-                if (!response.isError()) {
-                    System.out.println("Got response. All set: " + response.getBody(AdvertiseResponse.class));
-                } else {
-                    System.out.println("Error happened: " + response.getError().getMessage());
+                try {
+                    if (!response.isError()) {
```

Comment:
I believe isError should not throw. APIs where checking for errors can throw an error are very bad design

Reply:
the `isError()` won't (at least shouldn't) throw but `System.out` (or whatever actual code a client can use here) could. hence the `try { ... } finally { response.release(); }` here - as an example of how the `response` should be released no matter what.

Reply:
Oh, perhaps add a comment about this